### PR TITLE
Addition of Edit Connection Command

### LINKED ISSUE #5158: Edit properties of a server without recreating the connection
PowerUser will modify server properties relatively often. Current experience is that user will need to reconnect and go to Advanced Properties. There is currently not a way to modify the properties of an existing connection like in SSMS.

Feature request is to have a way to modify advanced properties on an existing connection without having to reconnect, especially when managing many connections.

### PR DESCRIPTION
**Pull request by smartguest**
This is simply a version of the AddServerAction except it loads the profile of the existing connection and allows the user to edit it instead, it can change the connection information of the server while it is running.
This PR fixes #5158

### CODE REVIEW

File: src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts [hunk 1]
```diff
@@ -79,6 +79,26 @@ export class RefreshAction extends Action {
...
+		label: string,
+		private _connectionProfile: ConnectionProfile,
+		@IConnectionManagementService private _connectionManagementService: IConnectionManagementService
+	) {
+		super(id, label);
+		this.class = 'edit-server-action';
+	}
+
+	public async run(element: ConnectionProfileGroup): Promise<boolean> {
+		await this._connectionManagementService.showConnectionDialog(undefined, undefined, this._connectionProfile);
```
Comment (kburtram) [inline src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts]: do you think this change will be sufficient to address the request?  For example, if I change things like the server or database won't I get two connection nodes in OE?

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts]: I agree. People can already essentially do this functionality by selecting the connection from the "Recently Used" list when creating a new connection (and then modifying the fields as appropriate) so while adding an action to do it helps make that a bit easier that doesn't address the main ask of the enhancement requests.

Comment (smartguest) [reply src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts]: It works for changing the optional name of the server connection, but I will need to test those features out

Comment (smartguest) [reply src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts]: Do I need to write a new connection widget to handle editing the server details? I don't think you can use the existing one

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts]: Why not? Sure the current widget doesn't currently support actually editing an existing connection in-place but the actual design of the dialog shouldn't need to change. You could just add a mode (edit/add/etc) that determines whether we copy and create a new connection or modify the connection passed in.

Comment (kburtram) [reply src/sql/workbench/contrib/objectExplorer/browser/connectionTreeAction.ts]: @smartguest I think connections should have a unique GUID to identify them.  So when the Connection Dialog edits an existing connection, that connection profile metadata should be updated.  So if the user changes the server or database name, the existing connection should be changed to reflect those values in Object Explorer, etc.  

Other UI, such as Query editors, notebooks, or dashboards that are using that are actively connection should not be impacted.  Only new connections created from that profile should use the updated values.  Please sync-up with me if the intended use-case isn't clear here.

**Pull request closed without merging**